aidial_adapter_openai/app_config.py

```diff
@@ -0,0 +1,103 @@
+import json
+import os
+from typing import Dict, List
+
+from pydantic import BaseModel
+
+from aidial_adapter_openai.constant import ChatCompletionDeploymentType
+from aidial_adapter_openai.utils.env import get_env_bool
+from aidial_adapter_openai.utils.json import remove_nones
+from aidial_adapter_openai.utils.log_config import logger
+from aidial_adapter_openai.utils.parsers import parse_deployment_list
+
+
+class ApplicationConfig(BaseModel):
+    MODEL_ALIASES: Dict[str, str] = {}
+    DALLE3_DEPLOYMENTS: List[str] = []
+    GPT4_VISION_DEPLOYMENTS: List[str] = []
+    MISTRAL_DEPLOYMENTS: List[str] = []
+    DATABRICKS_DEPLOYMENTS: List[str] = []
+    GPT4O_DEPLOYMENTS: List[str] = []
+    GPT4O_MINI_DEPLOYMENTS: List[str] = []
+    AZURE_AI_VISION_DEPLOYMENTS: List[str] = []
+    API_VERSIONS_MAPPING: Dict[str, str] = {}
+    COMPLETION_DEPLOYMENTS_PROMPT_TEMPLATES: Dict[str, str] = {}
+    DALLE3_AZURE_API_VERSION: str = "2024-02-01"
+    NON_STREAMING_DEPLOYMENTS: List[str] = []
+    ELIMINATE_EMPTY_CHOICES: bool = False
+
+    def get_chat_completion_deployment_type(
+        self, deployment_id: str
+    ) -> ChatCompletionDeploymentType:
+        if deployment_id in self.DALLE3_DEPLOYMENTS:
+            return ChatCompletionDeploymentType.DALLE3
+        elif deployment_id in self.GPT4_VISION_DEPLOYMENTS:
+            return ChatCompletionDeploymentType.GPT4_VISION
+        elif deployment_id in self.MISTRAL_DEPLOYMENTS:
+            return ChatCompletionDeploymentType.MISTRAL
+        elif deployment_id in self.DATABRICKS_DEPLOYMENTS:
+            return ChatCompletionDeploymentType.DATABRICKS
+        elif deployment_id in self.GPT4O_DEPLOYMENTS:
+            return ChatCompletionDeploymentType.GPT4O
+        elif deployment_id in self.GPT4O_MINI_DEPLOYMENTS:
+            return ChatCompletionDeploymentType.GPT4O_MINI
+        else:
+            return ChatCompletionDeploymentType.GPT_TEXT_ONLY
+
+    @classmethod
+    def from_env(cls) -> "ApplicationConfig":
+        def _parse_env_deployments(deployments_key: str) -> List[str] | None:
+            return parse_deployment_list(os.getenv(deployments_key)) or None
```

Comment:
`parse_deployment_list` never returns None, but it should when the env var is unset.

Reply:
Fixed, removed parse_deployment_list from utils, moved all logic here